onnxruntime/core/optimizer/selectors_actions/selector_action_transformer.h

```diff
@@ -9,14 +9,18 @@
 namespace onnxruntime {
 
 class Graph;
+class GraphViewer;
 class Node;
 
 #if !defined(ORT_MINIMAL_BUILD)
+
 // Base class for a selector which checks for a match and returns the set of nodes involved.
 struct NodeSelector {
   // Select one or more nodes for an Action to process if the constraints are satisfied.
   // `selection` should not be set if this returns false
-  virtual bool Select(Graph& graph, const Node& node, std::unique_ptr<NodesToOptimize>& selection) const = 0;
+  virtual bool Select(const GraphViewer& graph_viewer, const Node& node,
```

Comment:
I've actually got one question regarding this shift for keeping track of selected nodes from NodesToOptimize struct to NodeIndexes. 

-For supporting qdq in nnapi ep, I originally passed a selected qdq group struct(representing by `NodesToOptimize` -like struct) in my draft. And this information is needed for example in GetCapability() when checking if a node/op is supported, as for nodes like qdq target node - conv, the weight/bias initializers are now obtained from the dq nodes in the qdq group.  Using NodeIndexes to keep track is perfectly fine, but it seems like having NodeIndexes,  in order to get the node, we need the graphviewer right? i.e. `graph_viewer.GetNode(NodeIndex)`?  
Method `IsNodeSupported()` carries a graph_viewer, but starting from `IsOpsupported()/IsOpSupportedImpl()`, it does not carry  a graph_viewer, (guess it's too expensive).  
-So is there any better suggestions/ways to get a node from NodeIndexes I am not aware of?

Reply:
If you are using for qdq scenario only (for NNAPI we only need qdq selectors), you can use the QDQ::BaseSelector instead of NodeSelector, for which I still keep the Select returns the QDQ::NodeGroup

Or you can use the NodesToOptimizeIndexes if you prefer, you can get the same thing as QDQ::NodeGroup, but will need add some helpers

Reply:
Yes. thanks for the reply. I understand this. But the QDQ::NodeGroup struct is essentially still vector of NodeIndex right?

I guess my question is if we are not using Node* anymore but using NodeIndex, how to get the actual node in IsOpsupported()?

Reply:
Use the `graph_viewer.GetNode(NodeIndex)` to get the `Node` and pass it down?